### PR TITLE
[plugin.video.psvue] 2017.5.17

### PR DESCRIPTION
**Pull request by eracknaphobia**
### Description

This plugin enables users to watch PS Vue streams through Kodi.
https://www.playstation.com/en-us/network/vue/

*Requires a paid for ps vue account
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.psvue/resources/lib/globals.py [hunk 1]
```diff
@@ -0,0 +1,682 @@
...
+import random
+import cookielib, urllib, urlparse
+import json
+import requests
+import time, calendar
+from datetime import date, datetime, timedelta
+
+
+def main_menu():
+    addDir('Timeline', 50, ICON)
```
Comment (enen92) [inline plugin.video.psvue/resources/lib/globals.py]: Probably you can add those to the strings.po as well?

File: plugin.video.psvue/resources/lib/globals.py [hunk 2]
```diff
@@ -0,0 +1,682 @@
...
+        else:
+            msg = "Something went wrong during login"
+            dialog = xbmcgui.Dialog()
+            ok = dialog.ok('Login Failed', msg)
+            sys.exit()
+
+
+def two_step_verification(ticket_uuid):
+    dialog = xbmcgui.Dialog()
+    code = dialog.input('Please enter your verification code', type=xbmcgui.INPUT_ALPHANUM)
```
Comment (enen92) [inline plugin.video.psvue/resources/lib/globals.py]: strings.po? :)

File: plugin.video.psvue/resources/lib/globals.py [hunk 3]
```diff
@@ -0,0 +1,682 @@
...
+
+    payload = 'authentication_type=two_step&ticket_uuid='+ticket_uuid+'&code='+code+'&client_id='+LOGIN_CLIENT_ID
+    r = requests.post(url, headers=headers, cookies=load_cookies(), data=payload, verify=VERIFY)
+    json_source = r.json()
+    save_cookies(r.cookies)
+
+    if 'npsso' in json_source:
+        npsso = json_source['npsso']
+        ADDON.setSetting(id='npsso', value=npsso)
+    elif 'error_description' in json_source:
```
Comment (enen92) [inline plugin.video.psvue/resources/lib/globals.py]: same here if possible

File: plugin.video.psvue/resources/lib/globals.py [hunk 4]
```diff
@@ -0,0 +1,682 @@
...
+        msg = json_source['error_description']
+        dialog = xbmcgui.Dialog()
+        ok = dialog.ok('Login Failed', msg)
+        sys.exit()
+    else:
+        msg = "Something went wrong during login"
```
Comment (enen92) [inline plugin.video.psvue/resources/lib/globals.py]: and here ^

**Comment by enen92** [comment]
@eracknaphobia you're free to merge it. My comments are not a blocker at all. It would be cool though if you can add those hardcoded strings to your addon translation files. That way, in the future, if you want to have translations from transifex they'd be actually possible.

Regards

**Comment by eracknaphobia** [comment]
Agreed, thanks for the review :-)

**Pull request merged**